mono/mini/mini-amd64.c

```diff
@@ -181,7 +181,8 @@ amd64_use_imm32 (gint64 val)
 }
 
 static void
-amd64_patch (unsigned char* code, gpointer target)
+amd64_patch_aligned (unsigned char* code, gpointer target, gboolean aligned)
+// "aligned" just generated and never run, vs. in the midst of running.
 {
```

Comment:
Why is this new argument needed ?

Reply:
We frequently patch during initial codegen when alignment doesn’t matter, code not yet running. And also patch later when running and alignment matters.

Reply:
Granted: which instructions patched when? I didn’t check. I can try always asserting & no new param.